Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -627,4 +627,84 @@ public struct NonBlockingFileIO: NIOSendable {
         }
     }
 
+    public func lstat(path: String, eventLoop: EventLoop) -> EventLoopFuture<stat> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            let pointer = UnsafeMutablePointer<stat>.allocate(capacity: 1)
+            defer {
+                pointer.deallocate()
+            }
+            try Posix.lstat(pathname: path, outStat: pointer)
+            return pointer.pointee
+        }
+    }
+
+    public func symlink(path: String, to destination: String, eventLoop: EventLoop) -> EventLoopFuture<Void> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            try Posix.symlink(pathname: path, destination: destination)
+        }
+    }
+
+    public func readlink(path: String, eventLoop: EventLoop) -> EventLoopFuture<String> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            let maxLength = 1026
+            let pointer = UnsafeMutablePointer<CChar>.allocate(capacity: maxLength)
```

Comment:
Prefer `String(unsafeUninitializedCapacity:initializingWith:)`

Reply:
we don't get a number a number bytes in the resulting pointer, we just expect that it will be `\0`-terminated